DirectPInvoke with NativeAOT on Windows x86 fails on SetCredentialsAttributesW due to mismatched signature size

### Description

When utilising `DirectPInvoke` on a 32-bit Windows system (x86) target with NativeAOT, the MSVC linker fails to resolve `SetCredentialsAttributesW` with `error LNK2001: unresolved external symbol _SetCredentialsAttributesW@24`. 
The underlying issue is that the internal BCL LibraryImport maps the native unsigned long parameters directly to C# long (8 bytes)? 

output of `dumpbin /exports` on x86 version of `secur32.lib`: 
```
File Type: LIBRARY

     Exports

       ordinal    name

                  ...
                  _SetContextAttributesA@16
                  _SetContextAttributesW@16
                  _SetCredentialsAttributesA@16
                  _SetCredentialsAttributesW@16
                  _SspiCompareAuthIdentities@16
                  _SspiCopyAuthIdentity@8
                  _SspiDecryptAuthIdentity@4
                  ...
```

### Reproduction Steps

1. Create a .NET console app.
2. Configure the project to publish via NativeAOT targeting Windows x86 (`win-x86`).
3. Enable direct P/Invoke static compilation mapped to `sspicli` via `DirectPInvoke` item groups in your `.csproj`:

```xml
<PropertyGroup>
  <PublishAot>true</PublishAot>
</PropertyGroup>

<ItemGroup>
  <DirectPInvoke Include="sspicli" />
</ItemGroup>
```

4. Run a workflow that references the internal SSPI context.
5. Attempt to publish the project:
```bash
dotnet publish -r win-x86
```

### Expected behavior

The BCL signature should conditionally compile or natively match a 4-byte width.

### Actual behavior

Compilation halts during the linking phase with the following native compiler error:
`error LNK2001: unresolved external symbol _SetCredentialsAttributesW@24`

### Regression?

_No response_

### Known Workarounds

1. Exclude sspicli from direct P/Invoke
2. Set `/ALTERNATENAME:_SetCredentialsAttributesW@24=_SetCredentialsAttributesW@16` linker arg

### Configuration

_No response_

### Other information

_No response_

--- a/src/libraries/Common/src/Interop/Windows/SspiCli/Interop.SSPI.cs
+++ b/src/libraries/Common/src/Interop/Windows/SspiCli/Interop.SSPI.cs
@@ -538,8 +538,8 @@ internal static partial class Interop
         [LibraryImport(Interop.Libraries.SspiCli, SetLastError = true)]
         internal static partial SECURITY_STATUS SetCredentialsAttributesW(
             in CredHandle handlePtr,
-            long ulAttribute,
+            uint ulAttribute,
             in SecPkgCred_ClientCertPolicy pBuffer,
-            long cbBuffer);
+            uint cbBuffer);
     }
 }

--- a/src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Windows.cs
+++ b/src/libraries/System.Net.Security/src/System/Net/Security/SslStreamPal.Windows.cs
@@ -317,9 +317,9 @@ namespace System.Net.Security
                 clientCertPolicy.pwszSslCtlStoreName = ptr;
                 Interop.SECURITY_STATUS errorCode = Interop.SspiCli.SetCredentialsAttributesW(
                             cred._handle,
-                            (long)Interop.SspiCli.ContextAttribute.SECPKG_ATTR_CLIENT_CERT_POLICY,
+                            (uint)Interop.SspiCli.ContextAttribute.SECPKG_ATTR_CLIENT_CERT_POLICY,
                             clientCertPolicy,
-                            sizeof(Interop.SspiCli.SecPkgCred_ClientCertPolicy));
+                            (uint)sizeof(Interop.SspiCli.SecPkgCred_ClientCertPolicy));
 
                 if (errorCode != Interop.SECURITY_STATUS.OK)
                 {

